### PR TITLE
refactor: NoticeBar - marquee.native, close #1800.

### LINKED ISSUE #1800: NoticeBar在iOS上运行正常, 在Android上不滚动显示是静止的
### Version
1.6.5

### Environment
react-native 0.47.1

### Reproduction link
[https://codepen.io/paranoidjk/pen/LWpaKe](https://codepen.io/paranoidjk/pen/LWpaKe)

### Steps to reproduce
<NoticeBar marqueeProps={{loop:true,style:{color:'#666666',fontSize:13}}}  icon={null}>
       Notice: The arrival time of incomes and transfers of Yu &#39;E Bao will be delayed during National Day.
</NoticeBar>

### What is expected?
Android能和iOS一样在字数超过范围时能滚动显示

### What is actually happening?
Android在使用NoticeBar的时候字数超过显示范围时静止的

### PR DESCRIPTION
**Pull request by zhang740**
- 将create-react-class写法改成class
- fix: android下onLayoutContainer晚于onLayout，并且onLayout触发频繁导致的问题
- fix: android下切分Text的方法动画性能不足，改换单Text，扩大包围View的方式
---
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1815)

### CODE REVIEW

File: components/notice-bar/Marquee.native.tsx [hunk 1]
```diff
@@ -1,113 +1,114 @@
...
+        ellipsizeMode="tail"
+        style={style}
+      >
+        {text}
+      </Text>;
 
     return (
       <View style={{ flex: 1, flexDirection: 'row', alignItems: 'center' }} onLayout={this.onLayoutContainer}>
-        <Animated.View style={{ flexDirection: 'row', left: this.state.left }}>
+        <Animated.View style={{ flexDirection: 'row', transform: [{ translateX: this.state.left }], width: maxWidth }}>
```
Comment (silentcloud) [inline components/notice-bar/Marquee.native.tsx]: 这里为啥要 transform？ transform 这种情况肯定会比 left 要差的

**Comment by silentcloud** [comment]
ci

File: components/notice-bar/Marquee.native.tsx (hunk 1 above)
Comment (silentcloud) [reply components/notice-bar/Marquee.native.tsx]: 如果用 left 可以，换成 left 性能会好点

**Comment by zhang740** [comment]
ci

**Pull request closed without merging**